### PR TITLE
Update Bellhop Documentation

### LINKED ISSUE #17: Documentation
**Is your feature request related to a problem? Please describe.**
We need to make sure all of the documentation for Bellhop is crisp and concise. 

**Describe the solution you'd like**
The team to set time aside and go through the documentation to make it more user-friendly.

### PR DESCRIPTION
**Pull request by DCMattyG**
Updated all docs to make the Bellhop project, deployment, and utilization clear.

Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,124 +39,16 @@ This project was born out of customer demand to fill a gap in Azure's ability to
...
-
-**_IT IS RECOMMENDED TO USE AVAILABLE SECURITY CONTROLS IN A PRODUCTION DEPLOYMENT_**
+- **[Welcome to Bellhop!](https://azure.github.io/bellhop/#/)**
 
 ## FAQ
 
 **Why would I use Bellhop?**
 
-You realize that by "turning down" your resources in Dev/Test environments you could save a lot of money. You also realize there is not currently an an easy way to scale service tiers on Azure PaaS services. Bellhop to the rescue!!!! Tag your resources with scale times and desired "simmer" settings and Bellhop will take care of the rest!
+You realize that by "turning down" your resources in Dev/Test environments you could save a lot on the cost of your infrastructure. You also realize there is not currently an easy way to scale between service tiers on Azure PaaS services. Bellhop to the rescue!!!! Tag your resources with scale times and desired service tier settings and let Bellhop take care of the rest!
```
Comment (NillsF) [inline README.md:51]: Remove Paas?

File: docs/README.md [hunk 2]
```diff
@@ -8,96 +8,81 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
+## Bellhop Overview and Architecture
+This project was born out of the desire to be resource conscious and to fill a gap in Azure's native ability to easily scale managed services between service tiers. Bellhop provides a highly extensible framework, that is completely serverless in design leveraging Azure Functions, Storage Queues, and resource tags. Users just need to tag their resources with the required tags (covered below) and the Engine function will then query the Graph API, use those tag values to determine which resources need to be scaled, to which tier, and when. The Engine then posts a scale message in a predefined Storage Queue, triggering the Scaler function to pull the message from the queue and begin processing the scale request. The Scaler function itself is very lightweight and it is only used to import and pass parameter values to the correct Scaler Module, which handles each scaling operation. Every Azure resource that will be scaled will need a dedicated Scaler Module. All that users needs to do to extend this solution to any new Azure resources is write a new Scaler Module for that service, the Engine and Scaler Function code should rarely change.
```
Comment (NillsF) [inline docs/README.md:12]: didn't we make this shorter?

**Review by CloudViking — APPROVED** [review]
- Group work, changes made together
- Looks good

File: docs/README.md [hunk 3]
```diff
@@ -8,96 +8,81 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
 Enter Azure Region to deploy to: westus2
 ```
 
-### Steps to update the Scaler-Trigger Function when adding new scaler modules:
-
+### Steps to update the Scaler Function when adding custom scaler modules:
 - Run `updateScaler.ps1` from project root
 
-The update script will ask user for a Resource Group Name, and then zip deploy the new updates to the Scaler-Trigger function deployed in the given resource group.
+The update script will ask user for a Resource Group name, and then zip deploy the updates to the Scaler function deployed in the given resource group.
```
Comment (NillsF) [inline docs/README.md:59]: user --> the user (add "the")

File: docs/README.md [hunk 4]
```diff
@@ -8,96 +8,81 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
-PS /User/git_repos/github/Azure/bellhop> ./updateScaler.ps1
+PS /User/github/Azure/bellhop> ./updateScaler.ps1
 Enter resource group name where function is deployed: bellhop-rg 
 ```
 
 ### Steps to tear down the deployment:
 - Run `removeBellhop.ps1` from project root
 
-The teardown script will ask user for a Resource Group Name, and then delete that resource group and all associated resources. 
+The teardown script will ask user for a Resource Group name, and then delete that Resource Group and all associated resources.
```
Comment (NillsF) [inline docs/README.md:70]: THE user

File: docs/engine/README.md [hunk 5]
```diff
@@ -1,19 +1,26 @@
 # Bellhop Engine Function
 The Bellhop Engine Function is the brains of the operation, and is responsible for orchestrating all of the operations Bellhop performs. 
 
-The engine is written in C#, and requires .NET Core 3.1 be installed in order to deploy this solution. By default the Engine will trigger and run on a schedule repeating every 5 minutes. When triggered the Engine will query Azure Graph API for any Azure resources, within the subscription, that have the `"resize-Enable": "true"` tag set. 
+The engine is written in C# (.NET Core 3.1). By default the Engine will trigger and run on a schedule repeating every 5 minutes. When triggered the Engine will query Azure Resource Graph API for any Azure resources, within the subscription, that have the `"resize-Enable": "True"` tag set. Using Azure Resource Graph avoid querying the ARM API, reducing the likelihood of API throttling.
```
Comment (NillsF) [inline docs/engine/README.md:4]: avoidS

File: docs/scaler/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Bellhop Scaler Function
+The Scaler Function is a very lightweight Powershell function that is triggered when a new message is posted to the defined storage queue. When this function is invoked it will parse the queue message item, import the necessary scaler module based on resource type, and finally invoke the scaling module using the parameters collected.
+
+## Bellhop Scaler Function Workflow
+- Message is recieved in the queue after being passed from the Engine
+- Scaler parses the Resource Type, and load the appropriate Scaler Module (Import function.psm1)
```
Comment (NillsF) [inline docs/scaler/README.md:6]: loadS

File: docs/scaler/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+- Resize parameters are passed to the Scaler Module function:
+    - Resize direction (Up/Down)
+    - Azure Resource Graph results
+    - Extracted Tag data (Save/Set state)
+- Entire resize operation happens within the Scaler Module
```
Comment (NillsF) [inline docs/scaler/README.md:11]: happens within --> is executed by

File: docs/scaler/modules/sqlelastic.md [hunk 8]
```diff
@@ -2,14 +2,20 @@
 This README will describe how to use Bellhop to help scale your Azure SQL ELastic Pool Resources.
 
 ## Scaling Considerations
-Scaling Azure SQL Elastic Pools effectively is a bit more challenging, as you really have to make sure to account for scaling between Vcore editions and DTU's. Each type will have different tag options and requirements.
+
```
Comment (NillsF) [inline docs/scaler/modules/sqlelastic.md:5]: why is there an empty line?

**Review by NillsF — CHANGES_REQUESTED** [review]
minor comments. let's fix them and then get this PR merged.

File: README.md (hunk 1 above)
Comment (DCMattyG) [reply README.md:51]: Done

File: docs/README.md (hunk 2 above)
Comment (DCMattyG) [reply docs/README.md:12]: Agreed that we'll leave this as-is.

File: docs/README.md (hunk 3 above)
Comment (DCMattyG) [reply docs/README.md:59]: Fixed

File: docs/README.md (hunk 4 above)
Comment (DCMattyG) [reply docs/README.md:70]: Fixed

File: docs/engine/README.md (hunk 5 above)
Comment (DCMattyG) [reply docs/engine/README.md:4]: Fixed

File: docs/scaler/README.md (hunk 6 above)
Comment (DCMattyG) [reply docs/scaler/README.md:6]: Fixed

File: docs/scaler/README.md (hunk 7 above)
Comment (DCMattyG) [reply docs/scaler/README.md:11]: Fixed

File: docs/scaler/modules/sqlelastic.md (hunk 8 above)
Comment (DCMattyG) [reply docs/scaler/modules/sqlelastic.md:5]: Fixed

**Review by NillsF — APPROVED** [review]
LGTM

**Pull request merged**